crates/ra_ide_api/src/display.rs

```diff
@@ -0,0 +1,112 @@
+//! This module contains utilities for rendering turning things into something
+//! that may be used to render in UI.
+use super::*;
+use std::fmt::{self, Display};
+use join_to_string::join;
+use ra_syntax::ast::{self, AstNode, NameOwner, VisibilityOwner, TypeParamsOwner};
+use std::convert::From;
+
+/// Contains information about a function signature
+#[derive(Debug)]
+pub struct FunctionSignature {
+    /// Optional visibility
+    pub visibility: Option<String>,
+    /// Name of the function
+    pub name: Option<String>,

```

Comment:
Should `Name` be optional? I think that for signature help we always have named functions correct? I actually haven't tested with a closure.

Reply:
Currently closures don't seem to get signature help so we could make `Name` non-optional

Reply:
That's a bug, we should have signature help for closures, so optional name is the right choice here.

Reply:
Awesome. In that case I think closure support could be a separate PR.